### ISSUE TITLE
Remove CPP code in favor of numba

### ISSUE THREAD

**Issue by cbespin** [SiLab-Bonn/beam_telescope_analysis#27]
Browsing the code, I noticed usage of Cython functions for histogramming and array operations. Maybe those can be ported to numba to remove the cython/cpp dependency, making this project pure python.
If this is considered useful, I am willing to do a speed comparison and provide a PR when I find the time.

**Comment by leloup314** [comment]
Sounds nice to me ;) Maybe you have the time to look at the [`pylandau` numba](https://github.com/SiLab-Bonn/pylandau/tree/numba )port (which is essential done) and check if this is fine? This uses AOT instead of a (probably) faster JIT approach

**Comment by cbespin** [comment]
Yes, I will (come time). The same can probably applied here, if it proves as an advantage.

**Comment by YannickDieter** [comment]
I like the idea. :)
